From 4791a4602bd108d56dce51604acf675c5fa7571d Mon Sep 17 00:00:00 2001
From: bluebunny <1812-bluebunny@users.noreply.gitlab.postmarketos.org>
Date: Mon, 27 Jan 2025 01:19:39 +0300
Subject: [PATCH] drivers: media: i2c: ov13b10: add device tree support  and
 dvdd/dovdd

Linux 7.2 handles all three regulators upstream; retain the OF match table.
---
 drivers/media/i2c/ov13b10.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/media/i2c/ov13b10.c b/drivers/media/i2c/ov13b10.c
index b0d34141a13a..e60d64cad5a9 100644
--- a/drivers/media/i2c/ov13b10.c
+++ b/drivers/media/i2c/ov13b10.c
@@ -1705,11 +1705,20 @@ static const struct acpi_device_id ov13b10_acpi_ids[] = {
 MODULE_DEVICE_TABLE(acpi, ov13b10_acpi_ids);
 #endif
 
+#ifdef CONFIG_OF
+static const struct of_device_id ov13b10_of_match[] = {
+	{ .compatible = "ovti,ov13b10" },
+	{ /* sentinel */ }
+};
+MODULE_DEVICE_TABLE(of, ov13b10_of_match);
+#endif
+
 static struct i2c_driver ov13b10_i2c_driver = {
 	.driver = {
 		.name = "ov13b10",
 		.pm = pm_ptr(&ov13b10_pm_ops),
 		.acpi_match_table = ACPI_PTR(ov13b10_acpi_ids),
+		.of_match_table = of_match_ptr(ov13b10_of_match),
 	},
 	.probe = ov13b10_probe,
 	.remove = ov13b10_remove,
-- 
2.55.0

